File: pkg/taucorder/service/health.go
package service

import (
	"context"

	"connectrpc.com/connect"
	pb "github.com/taubyte/tau/pkg/taucorder/proto/gen/taucorder/v1"
)

func (ts *healthService) Ping(ctx context.Context, req *connect.Request[pb.Empty]) (*connect.Response[pb.Empty], error) {
	return connect.NewResponse(&pb.Empty{}), nil
}

File: pkg/taucorder/clients/js/src/Health.ts
import { Transport } from "@connectrpc/connect";
import { RPCClient } from "./HealthClient";
import { Empty, Node } from "../gen/taucorder/v1/common_pb";

export class Health {
  private client: RPCClient;

  constructor(transport: Transport) {
    this.client = new RPCClient(transport);
  }

  /**
   * Ping the health service
   * @returns Empty response
   */
  async ping() {
    await this.client.ping(new Empty());
  }
} 